supabase/functions/cron_email/index.ts

```diff
@@ -0,0 +1,152 @@
+import { serve } from 'https://deno.land/std@0.199.0/http/server.ts'
+import { getEnv, methodJson, sendRes } from '../_utils/utils.ts'
+import type { BaseHeaders } from '../_utils/types.ts'
+import { trackEvent } from '../_utils/plunk.ts'
+import { supabaseAdmin } from '../_utils/supabase.ts'
+
+// This is required propoably due to this https://github.com/supabase/postgrest-js/issues/408
+interface AppWithUser {
+  app_id: string
+  name: string | null
+  user_id: {
+    email: string
+  }
+}
+
+const thresholds = {
+  updates: [
+    100,
+    1000,
+    10000,
+  ],
+  failRate: [
+    0.80,
+    0.90,
+    0.95,
+  ],
+  appOpen: [
+    500,
+    1500,
+    5000,
+  ],
+}
+
+const funComparisons = {
+  updates: [
+    'a cupcake to every student in a small school!',
+    'a pizza to every resident of a small town!',
+    'a burger to everyone in a big city!',
+  ],
+  failRate: [
+    'Even cats don\'t land on their feet that often!',
+    'That\'s a success rate higher than the average pass rate of a tough university exam!',
+    'That\'s a success rate that even the best basketball players would envy!',
+  ],
+  appOpen: [
+    'Your app was opened more times than a popular local bakery\'s door!',
+    'Your app was more popular than the latest episode of a hit TV show!',
+    'Your app was opened more times than a blockbuster movie on its opening weekend!',
+  ],
+}
+
+function getFunComparison(comparison: 'updates' | 'failRate' | 'appOpen', stat: number): string {
```

Comment:
you could do something like 
```suggestion
function getFunComparison(comparison: ComparisonEnum, stat: number): string {
```

or even 
```suggestion
function getFunComparison(comparison: keyof funComparisons, stat: number): string {
```

This would likely improve readability + avoid code repetition between key names

Reply:
Keyof will not work
![image](https://github.com/Cap-go/capgo/assets/50914789/43aa4e43-1b10-42fe-8a00-3384b34729d6)

As for enum, I do not think we really need it, this `funComparisons` are only used once (in this `getFunComparison` fn)

Reply:
I messed up here, the right `keyof typeof funComparisons`; can you check if it works?

Reply:
Yes it does, thanks for this suggestion

Reply:
Changed [here](https://github.com/Cap-go/capgo/pull/359/commits/6092f16622d27cf99f4ef74fee974bbd583e5c94)